opm/autodiff/WellInterface.hpp

```diff
@@ -312,8 +314,14 @@ namespace Opm
         bool checkRateEconLimits(const WellEconProductionLimits& econ_production_limits,
                                  const WellState& well_state) const;
 
+        bool underPredictionMode() const;
```

Comment:
This function turned out to be not used in this PR, since I was struggling to decide how to split PRs. It will be used later. Please let me know if you want to remove it from this PR.

Reply:
Do what is least work for you.